packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -650,6 +643,53 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
         }
     }
 
+    /**
+     * Verifies that the blob with given id is valid, i.e., it has not been garbage collected. If the blob is GC'd,
+     * log an error and throw if necessary.
+     */
+    private verifyBlobValidity(blobId: string) {
+        /**
+         * A blob can be in one of the following states:
+         * 1. "deleted" - It has been deleted by garbage collection sweep phase.
+         * 2. "tombstoned" - It is ready for deletion but sweep phase isn't enabled and tombstone feature is enabled.
+         * 3. "valid" - It has not been deleted or tombstoned.
+         */
+        let state: "valid" | "tombstoned" | "deleted" = "valid";
+        if (this.isBlobDeleted(this.getBlobGCNodePath(blobId))) {
+            state = "deleted";
+        } else if (this.tombstonedBlobs.has(blobId) ) {
+            state = "tombstoned";
+        }
+
+        if (state === "valid") {
+            return;
+        }
+
+        // If the blob is deleted or throw on tombstone load is enabled, throw an error which will fail any attempt
+        // to load the blob.
+        const shouldFail = state === "deleted" || this.throwOnTombstoneLoad;
+        const request = { url: blobId };
+        const error = responseToException(createResponseError(
+            404,
+            "Blob was deleted",
+            request,
+            state === "tombstoned" ? { [TombstoneResponseHeaderKey]: true } : undefined,
+        ), request);
+        sendGCUnexpectedUsageEvent(
+            this.mc,
+            {
+                eventName: state === "tombstoned" ? "GC_Tombstone_Blob_Requested" : "GC_Deleted_Blob_Requested",
```

Comment:
Are we calling it GC_Deleted or GC_Sweep - which do you prefer? Personally, I thought it would be Sweep as that's what we've been calling it, but it doesn't matter to me.

Reply:
I think deleted makes more sense from a user perspective - some object was deleted and used is pretty clear. Sweep is more of an implementation detail, i.e., we use mark and sweep phases to eventually delete things.

Reply:
Sounds good to me. GC_Deleted it is